devops/pipeline/stages/e2etest_cloud.yaml

```diff
@@ -0,0 +1,210 @@
+# E2E Cloud Test Stage Definition
+# Parameters:
+#  - (Required) target [object]: The cloud test target object, defined as follows:
+#       distroName [string]: The name to give this vm cloud distro
+#       image_publisher [string]: The image publisher to use. See https://docs.microsoft.com/en-us/azure/virtual-machines/linux/cli-ps-findimage
+#       image_offer [string]: The image offer to use. See https://docs.microsoft.com/en-us/azure/virtual-machines/linux/cli-ps-findimage
+#       image_sku [string]: The image sku to use. See https://docs.microsoft.com/en-us/azure/virtual-machines/linux/cli-ps-findimage
+#       image_version [string]: the image version to use. See https://docs.microsoft.com/en-us/azure/virtual-machines/linux/cli-ps-findimage
+#       packagePattern [string]: Package to install on the target host from the Package Build pipeline (example '**/*focal_x86_64.deb')
+#       device_id [string]: The device ID to use on provisioning the device to the IoT Hub
+#       pre_script [string]: The script to run on the target machine BEFORE OSConfig is installed
+#  - (Optional) skipTeardown [boolean]: Leave test infrastructure in-place - default: false
+#  - (Optional) twinTimeout [number]: Timeout for twin updates - default: 90 seconds
+#  - (Optional) postScript [string]: bash script to perform post osconfig install (before agent start)
+#  - (Optional) testExpression [string]: the test expression to use to filter the tests (see https://docs.microsoft.com/dotnet/core/testing/selective-unit-tests?pivots=mstest) - default: all tests
+#  - (Optional) sourceBranch [string]: package branch to use - default: Uses source branch
+# 
+# Find different VM variants on the Azure Marketplace
+# See https://docs.microsoft.com/en-us/azure/virtual-machines/linux/cli-ps-findimage
+
+parameters:
+- name: target
+  displayName: The Cloud VM to target
+  type: object
+- name: skipTeardown
+  displayName: Leave test infrastructure in-place
+  type: boolean
+  default: false
+- name: twinTimeout
+  displayName: Timeout for twin updates
+  type: number
+  default: 90
+# - name: PRE_SCRIPT
+#   displayName: 'Shell script to execute before osconfig install'
+#   type: string
+#   default: ' '
+- name: postScript
+  displayName: 'Shell script to execute after osconfig install'
+  type: string
+  default: ' '
+- name: testExpression
+  type: string
+  default: ' '
+- name: sourceBranch
```

Comment:
Is this different than the branch that the E2E Pipeline run is triggered on? Shouldn't the sources always be taken from the package build on the same branch?

Reply:
yes, and thats why that is the default behavior. If you are testing new scenarios on existing binaries however (whats is already on main), this is where it comes in useful. In order to trigger the tests we require the topic branch to have packages available.

Reply:
Ahh I see, thanks for the explanation 👍🏻